File: frontend/src/services/attendanceService.js
import api from './api'

export const attendanceService = {
  mark: (payload) => api.post('/attendance/mark', payload),
  getByClassAndDate: (classId, date) => api.get(`/attendance/class/${classId}`, { params: { date } }),
  getByStudent: (studentId) => api.get(`/attendance/student/${studentId}`),
}

File: backend/src/main/java/com/school/service/AttendanceService.java
package com.school.service;

import com.school.dto.request.AttendanceMarkRequest;
import com.school.dto.response.AttendanceResponse;

import java.time.LocalDate;
import java.util.List;

public interface AttendanceService {
    List<AttendanceResponse> markAttendance(AttendanceMarkRequest request, String markedByTeacherId);
    List<AttendanceResponse> getByClassAndDate(String classId, LocalDate date, String subjectId);
    List<AttendanceResponse> getByStudent(String studentId);
}